### PR TITLE
Change RBAC to enable cluster scope

### PR DESCRIPTION
**Pull request by chinhuang007**
Change RBAC to enable cluster scope as the default mode
- role
- role binding
- install default

Signed-off-by: Chin Huang <chhuang@us.ibm.com>

#### Motivation
Make cluster scope as the default mode for the modelmesh serving controller, to be consistent with kserve

#### Modifications
RBAC roles and role binding
Install default is changed to cluster scope

#### Result
Controller will have cluster role and default installation is in cluster scope mode

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -13,14 +13,15 @@
...
-kind: Role
+kind: ClusterRole
 metadata:
   name: controller-role
 rules:
   - apiGroups:
       - ""
     resources:
       - configmaps
+      - namespaces
```
Comment (pvaneck) [inline config/rbac/role.yaml:24]: I think you may have forgotten the kubebuilder annotation (`+kubebuilder:rbac`) for the `namespaces` resource. Perhaps add it as part of the servingruntime_controller.go rbac annotations?

Comment (chinhuang007) [reply config/rbac/role.yaml]: Yes, updated role.yaml and put annotations in service_controller.go.

**Comment by pvaneck** [comment]
Thanks @chinhuang007 
/lgtm

**Review by njhill — APPROVED** [review]
Thanks @chinhuang007, LGTM

We may want to include in the config alternative RBAC config for the namespace-scope case (perhaps in separate subdir), but can always do that as a follow-on

**Pull request merged**